django/db/migrations/graph.py

```diff
@@ -126,33 +126,66 @@ def clear_cache(self):
                 self.node_map[node].__dict__.pop('_descendants', None)
             self.cached = False
 
-    def forwards_plan(self, node):
+    def forwards_plan(self, target):
         """
         Given a node, returns a list of which previous nodes (dependencies)
         must be applied, ending with the node itself.
         This is the list you would follow if applying the migrations to
         a database.
         """
-        if node not in self.nodes:
-            raise NodeNotFoundError("Node %r not a valid node" % (node, ), node)
+        if target not in self.nodes:
+            raise NodeNotFoundError("Node %r not a valid node" % (target, ), target)
         # Use parent.key instead of parent to speed up the frequent hashing in ensure_not_cyclic
-        self.ensure_not_cyclic(node, lambda x: (parent.key for parent in self.node_map[x].parents))
+        self.ensure_not_cyclic(target, lambda x: (parent.key for parent in self.node_map[x].parents))
         self.cached = True
-        return self.node_map[node].ancestors()
-
-    def backwards_plan(self, node):
+        node = self.node_map[target]
+        try:
+            return node.ancestors()
+        except RuntimeError:
+            # fallback to iterative dfs
```

Comment:
Can you issue a `RuntimeWarning` here telling the user that Django falls back and it might be time to clean up using `squashmigrations` and link to https://docs.djangoproject.com/en/dev/topics/migrations/#squashing-migrations.